Stripe/STPPaymentIntentParams.m

```diff
@@ -183,4 +184,20 @@ + (nonnull NSDictionary *)propertyNamesToFormFieldNamesMapping {
              };
 }
 
+#pragma mark - Utilities
+
++ (BOOL)isClientSecretValid:(NSString *)clientSecret {
+    static dispatch_once_t onceToken;
+    static NSRegularExpression *regex = nil;
+    dispatch_once(&onceToken, ^{
+        regex = [[NSRegularExpression alloc] initWithPattern:@"^pi_[^_]+_secret_[^_]+$"
```

Comment:
This will match any non-underscore character, so `pi_    _secret_  ` will match.

I ended up using `^pi_([a-zA-Z0-9])+_secret_([a-zA-Z0-9])+$`

Reply:
that seems reasonable to me, I'm unsure the source of truth here but stripe.js just uses the `[^_]` https://git.corp.stripe.com/stripe-internal/stripe-js-v3/blob/733bc8be/src/payment_intent/shared/index.js#L211